### ISSUE TITLE
Incorrect bucket name 

### ISSUE THREAD

**Issue by amuresia** [aws-amplify/amplify-cli#3741]
**Describe the bug**
I am adding a storage resource and the name of the created bucket has the env added twice.
**Amplify CLI Version**
`4.16.1` and `4.17.1`

**To Reproduce**
amplify add storage
give the bucket a name like myapp-storage

**Expected behavior**
I expect a bucket called `myapp-storage-dev` to be created (when the checked out amplify env is dev)

**Actual behaviour**
A bucket called `myapp-storagedev-dev` is created

**Desktop (please complete the following information):**
 - OS: Mac
 - Node Version. `v10.16.0`

**Comment by kaustavghosh06** [comment]
@amuresia We weren't able to reproduce this in us-east-1 region. Which region are you deploying to?

**Comment by amuresia** [comment]
Hi @kaustavghosh06, I am deploying in eu-west-2 (London)

**Comment by akshbhu** [comment]
Hi @amuresia 

According to AWS bucketName docs : **Bucket names must be unique across all existing bucket names.** To avoid this problem , if the user has same bucketName and env name in some earlier project, we append the a part of stackName to maintain uniqueness in the bucketNames.

Thats why there is one extra dev append to your bucketName.You can find the bucketName in awsexports.js file and you can use the file for rest bucket related operations.

If you are stuck on something with this naming convention, do let me know

**Comment by amuresia** [comment]
Hi @akshbhu, thanks for your reply.
I understand that the S3 bucket names have to be unique.
To recap, I was trying to add storage for my project and expected the `myapp-storage-dev` to be created, instead I got `myapp-storagedev-dev`.
I had a previous backend that was deployed with Amplify last year and `myapp-storage-dev` was created successfully. I destroyed that environment though, and manually removed the bucket from S3.
The bucket was not visible in S3 and furthermore after deploying test and prod environments with Amplify, I got `myapp-storagetest-test` and `myapp-storageprod-prod`, which definitely never existed in S3.
I can PM you the actual name of `myapp` if it helps. The issue is not a blocker for me but I thought I should understand why it's happening and see if there is a way to rollout a fix to prod after confirming that it works in dev and test environments.
